sdb/commands/zfs/dbuf.py

```diff
@@ -0,0 +1,123 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import argparse
+from typing import Iterable
+
+import drgn
+import sdb
+
+
+class DslDir():
+
+    @staticmethod
+    def name(dd):
+        pname = ''
+        if dd.dd_parent:
+            pname = DslDir.name(dd.dd_parent) + '/'
+        return pname + dd.dd_myname.string_().decode("utf-8")
+
+
+class Dataset():
+
+    @staticmethod
+    def name(ds):
+        if not ds:
+            return 'MOS'
+        name = DslDir.name(ds.ds_dir)
+        if not ds.ds_prev:
+            sn = ds.ds_snapname.string_().decode("utf-8")
+            if len(sn) == 0:
+                sn = '%UNKNOWN_SNAP_NAME%'
+            name += '@' + sn
+        return name
+
+
+class Objset():
+
+    @staticmethod
+    def name(os):
+        return Dataset.name(os.os_dsl_dataset)
+
+
+class Dbuf(sdb.Locator, sdb.PrettyPrinter):
```

Comment:
Can we add some documentation in the form of a docstring here? Then we can get access to it via the new `help` command.

Reply:
I switched to the `_init_parser()` way of setting up the parser, which got me a better `help dbuf` message.  Could you point me to an example of how to write the docstring and typical contents?

Reply:
We don't have any examples yet.. but it should work out well if you follow a convention like this:
```
"""
One-line summary here

DESCRIPTION
    Multi-line summary here.

EXAMPLES
    Multi-line section with examples.
```

The one line summary is required/expected, but apart from that, the rest of the docstring can be any format. The `SUMMARY` section that `help` displays is currently formatted with the body of that section having 4 spaces for indentation, so if the sections in your docstring adhere to that same 4 space indentation, it'll look consistent with the `SUMMARY` section.

I'm open to other opinions/suggestiongs on the formatting, and/or the sections that we should try to populate. It's relatively easy to change things now, since we haven't added any docstrings yet.